### ISSUE TITLE
incompatible with pnpm (native federation)

### ISSUE THREAD

**Issue by markusahrweileramcon** [angular-architects/module-federation-plugin#593]
When i create a fresh workspace with pnpm and i want to run any application, I get the following error:

```
An unhandled exception occurred: Package subpath './src/builders/application' is not defined by "exports" in \node_modules\.pnpm\node_modules\@angular\build\package.json
```

**Comment by jerousseau** [comment]
You need to replace the path in "/node_modules/@angular-architects/native-federation/src/utils/patch-angular-build.js"

```
-    const packagePath = path.join(workspaceRoot, 'node_modules/@angular/build/package.json');
-    const privatePath = path.join(workspaceRoot, 'node_modules/@angular/build/private.js');
+    const packagePath = path.join(workspaceRoot, 'node_modules/.pnpm/node_modules/@angular/build/package.json');
+    const privatePath = path.join(workspaceRoot, 'node_modules/.pnpm/node_modules/@angular/build/private.js');
```

and apply the patch :
`node node_modules/@angular-architects/native-federation/src/patch-angular-build.js`

**Comment by amipei** [comment]
If you use pnpm, you can set public-hoist-pattern=@angular/build in .npmrc file.
public-hoist-pattern=*@angular/build*

My guess is that pnpm and npm handle dependencies differently (yarn should be similar)

**Comment by diandsonc** [comment]
To use `yarn` you need to add the `@angular/build` package to `devDependencies`.

**Comment by manfredsteyer** [comment]
This will be resolved with Version 18.1 that lands very soon.
